### PR TITLE
Remove TextureImage with preference for RGB565.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This should help us on our way to color correctness. In another patch, I will attempt to roll skia. That should fix the performance/artifact issues related to tiling of larger images.

@brianosman: Is the `SkImage::preroll` call sufficient to realize the texture upload immediately? We are on the resource thread so this is really the best time to perform upload operations.

cc @mattsarett since this is required for color correctness.

### CODE REVIEW

**Comment by jason-simmons** [comment]
LGTM

**Comment by chinmaygarde** [comment]
Hmm, I am just attempting a Skia roll and `SkImage::preroll` seems to have gone away. That was short lived :(

**Comment by brianosman** [comment]
Yes, preroll got removed recently, because it wasn't as helpful as it
seemed (depending on the eventual usage of the image, you could end up
re-uploading anyway). There was a similar function "makeTextureImage" that
I also removed, for similar reasons. I'm open to bringing back
makeTextureImage with a different signature. As it stands, we don't have a
good way of ensuring that things are uploaded unless the client does the
backend work and hands us a GrBackendTextureDesc.

You could use SkImage::MakeTextureFromPixmap, although I'm also planning to
remove that. ;) I guess if you start using it, I'll have no choice but to
write makeTextureImage again. :)

**Comment by chinmaygarde** [comment]
I understand why preroll had to go away. I am assuming MakeTextureImage does not keep a host side copy of the image data. In that case, it lines up really well with the TextureImage stuff being removed in this patch. So thank you for bringing that back.

File: lib/ui/painting/image_decoding.cc [hunk 1]
```diff
@@ -41,14 +39,18 @@ sk_sp<SkImage> DecodeImage(std::vector<uint8_t> buffer) {
   if (generator == nullptr)
     return nullptr;
 
-  // First, try to create a texture image from the generator.
```
Comment (brianosman) [inline lib/ui/painting/image_decoding.cc]: I think all of this logic can now be:

// Matt has an upcoming CL that changes this next line to:
// auto srgb = SkColorSpace::MakeSRGB();
auto srgb = SkColorSpace::MakeNamed(SkColorSpace::kSRGB_Named);

auto image = SkImage::MakeFromEncoded(std::move(sk_data));
image = image->makeTextureImage(context, srgb.get());
return image;

**Comment by chinmaygarde** [comment]
I am landing this with the `Preroll` in place. Have another patch that roll skia and gets rid of the `Preroll`.

**Comment by chinmaygarde** [comment]
Closing in favor of https://github.com/flutter/engine/pull/3434

**Pull request closed without merging**